### DISCUSSION TITLE
[SQL Server] Executing `ALTER ROLE` does nothing

### DISCUSSION THREAD

**Discussion by SimeonStoykovQC** [sqlalchemy/sqlalchemy#9311] [Usage Questions] [answered]
I would like to execute [`ALTER ROLE` statements](https://learn.microsoft.com/en-us/sql/t-sql/statements/alter-role-transact-sql?view=sql-server-2017) programmatically, but doing so through SQLAlchemy doesn't seem to be doing anything. I am not even getting errors if I put wrong role names or member names; it seems that the query is just not executed in the database:

```python
import sqlalchemy as sa
eng = sa.create_engine(...)
eng.connect().execute(sa.text("USE tempdb; ALTER ROLE nonexistent ADD MEMBER nonexistent;"))  # doesn't raise
```

When executing this through a database prompt I am getting the expected errors that the role or the member  don't exist.

Is this behavior intended, is it possible to achieve this, or is there a better way outside of sqlalchemy?

**Comment by CaselIT** [comment, marked as answer]
Hi,

Depending on the dbapi in use multiple statement in a single execute may not be allowed, so it may just execute up to the the `;`.
Also you are not committing that statement. I don't know if it's needed, but it may also be a factor here.

Try:
```py
with eng.begin() as conn:
  conn.execute(sa.text("USE tempdb"))
  conn.execute(sa.text("ALTER ROLE nonexistent ADD MEMBER nonexistentb"))
```

**Reply by SimeonStoykovQC** [reply]
Thanks! Using one execution per statement worked fine 🚀
